test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/ReadOnlyTagHelperAttributesTest.cs

```diff
@@ -0,0 +1,643 @@
+                    { new[] { C, B, A3, A }, A2, -1 },
+                    { new[] { C, B, A3, A }, new TagHelperAttribute(A3.Name, A2.Value), -1 },
+                    { new[] { C, B, A3, A }, new TagHelperAttribute("aname_not", "random value"), -1 },
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(IndexOfData))]
+        public void IndexOf_ReturnsExpectedResult(
+            IEnumerable<IReadOnlyTagHelperAttribute> initialAttributes,
+            IReadOnlyTagHelperAttribute attributeToLookup,
+            int expected)
+        {
+            // Arrange
+            var attributes = new ReadOnlyTagHelperAttributes<IReadOnlyTagHelperAttribute>(initialAttributes);
+
+            // Act
+            var index = attributes.IndexOf(attributeToLookup);
+
+            // Assert
+            Assert.Equal(expected, index);
+        }
+
+        public static TheoryData TryGetAttributeData
+        {
+            get
+            {
+                var A = new TagHelperAttribute("AName", "AName Value");
+                var A2 = new TagHelperAttribute("aname", "AName Second Value");
+                var A3 = new TagHelperAttribute("AName", "AName Third Value");
+                var B = new TagHelperAttribute("BName", "BName Value");
+                var C = new TagHelperAttribute("CName", "CName Value");
+
+                return new TheoryData<
+                    IEnumerable<IReadOnlyTagHelperAttribute>, // initialAttributes
+                    string, // nameToLookup
+                    IReadOnlyTagHelperAttribute, // expectedAttribute
+                    bool> // expectedResult
+                {
+                    { new[] { A }, "AName", A, true },
+                    { new[] { A }, "AnAmE", A, true },
+                    { new[] { A, B }, "AName", A, true },
+                    { new[] { A, B }, "AnAmE", A, true },
+                    { new[] { A, B }, "BName", B, true },
+                    { new[] { A, B }, "BnAmE", B, true },
+                    { new[] { A, B, C }, "BName", B, true },
+                    { new[] { A, B, C }, "bname", B, true },
+                    { new[] { A, B, C }, "CName", C, true },
+                    { new[] { A, B, C }, "cnamE", C, true },
+
+                    // Multiple elements same name
+                    { new[] { A, B, A2, C }, "AName", A, true },
+                    { new[] { A, B, A2, C }, "aname", A, true },
+                    { new[] { B, A2, A }, "aname", A2, true },
+                    { new[] { B, A2, A, C }, "AName", A2, true },
+                    { new[] { A, A3 }, "AName", A, true },
+                    { new[] { A3, A }, "aname", A3, true },
+                    { new[] { A, A2, A3 }, "AName", A, true },
+                    { new[] { C, B, A3, A }, "AName", A3, true },
+
+                    // Null expected lookups
+                    { new[] { A }, "_AName_", null, false },
```

Comment:
not a major coverage issue but should consistently include `"__similar to a match__"` and `"completely different"` names in these data sets.